app/models/miq_expression/field.rb

```diff
@@ -0,0 +1,46 @@
+class MiqExpression::Field
+  FIELD_REGEX = /
+(?<model_name>[[:upper:]][[:alnum:]]*)
+\.?(?<associations>[a-z_\.]+)*
+-(?<column>[a-z]+(_[a-z]+)*)
+/x
+
+  def self.parse(field)
+    match = FIELD_REGEX.match(field)
+    new(match[:model_name].constantize, match[:associations].to_s.split("."), match[:column])
+  end
+
+  attr_reader :model, :associations, :column
+
+  def initialize(model, associations, column)
+    @model = model
+    @associations = associations
+    @column = column
+  end
+
+  def table_name
+    target.table_name
+  end
+
+  def date?
+    column_type == :date
+  end
+
+  def datetime?
+    column_type == :datetime
+  end
+
+  private
+
+  def target
```

Comment:
@imtayadeway That original code handles getting the table name for associations in addition to the main model class. 

For example, getting the table names for fields in an expression on the HOst class:
Operating system associated to a host:
```
2.2.3 :071 > MiqExpression.get_sqltable("Host.operating_system")
 => "operating_systems" 
```
Compared to directly on a host:
```
2.2.3 :073 > MiqExpression.get_sqltable("Host")
 => "hosts" 
```